pulp_ansible/app/tasks/collections.py

```diff
@@ -422,15 +424,33 @@ async def _fetch_collection_version_metadata(self, api_version, collection_versi
     async def _add_collection_version(self, api_version, collection_version_url, metadata):
         """Add CollectionVersion to the sync pipeline."""
         url = metadata["download_url"]
-
         collection_version = CollectionVersion(
             namespace=metadata["namespace"]["name"],
             name=metadata["collection"]["name"],
             version=metadata["version"],
         )
+        cv_unique = attrgetter("namespace", "name", "version")(collection_version)
+        if cv_unique in self.already_synced:
+            return
+        self.already_synced.add(cv_unique)
```

Comment:
@bmbouter Here is the dictionary (set) check for adding the CV to the pipeline. Tell me if I need to have this before the CollectionVersion object is created. I did it this way since I didn't like how long the tuple `(metadata["namespace"]["name"], metadata["collection"]["name"], metadata["version"])` was, but I can change it to that and remove the extra import I added if you want.

Reply:
This is good how you've written it. +1 `attrgetter`

Reply:
It also more DRY in case how the `namespace`, `name`, and `version` is read in from the metadata in the future.